x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/components/post_install_cloud_formation_modal.tsx

```diff
@@ -41,9 +45,19 @@ export const PostInstallCloudFormationModal: React.FunctionComponent<{
 
   const cloudFormationProps = getCloudFormationPropsFromPackagePolicy(packagePolicy);
 
+  const fleetServerHostId = agentPolicy?.fleet_server_host_id;
+  const { data } = useGetFleetServerHosts();
+
+  const fleetServerHosts = data ? data.items ?? [] : [];
+
+  const selectedHost = fleetServerHosts.find((host) => host.id === fleetServerHostId);
+
+  const fleetServerUrl = selectedHost?.host_urls?.[0];
```

Comment:
This logic is duplicated on `cloud_formation_instructions.tsx`, let's abstract this common logic into a utility function under `x-pack/plugins/fleet/public/services` folder that we pass the `fleetServerHostId` and it returns the `fleetServerUrl`.

Reply:
we don't have the duplicated logic anymore as for setup instructions we are reusing the fleet host that is available in the component already.